scripts/setBuildVersion.sh

```diff
@@ -1,16 +1,30 @@
 #!/bin/bash
-if [ "$CONFIGURATION" == "Debug" ]
+# Use git commit short SHA for Debug/AdHocDistribution configuration (e.g. TestFlight)
+if [ "$CONFIGURATION" == "Debug" -o "$CONFIGURATION" == "AdHocDistribution" ]
 then
     # Get current git commit short SHA
     gitCommitSHA=$(git rev-parse --short HEAD)
     # Check if the git working copy is dirty
-    if test -n "$(git status --porcelain)"
+    if [ -n "$(git status --porcelain)" ]
     then
         # Append a plus character to the git commit short SHA to indicate dirty changes in build
         gitCommitSHA+="+"
     fi
-    # Set bundle build version to current git commit short SHA with dirty indicator
+fi
+
+# Use git commit count for Master branch and AppStoreDistribution configuration (e.g. App Store release)
+currentGitBranchName=$(git rev-parse --abbrev-ref HEAD)
+if [ "$currentGitBranchName" == "master" -a "$CONFIGURATION" == "AppStoreDistribution" ]
+then
+    # Get current git commit count
+    gitCommitSHA=$(git rev-list HEAD | wc -l | tr -d ' ')
+fi
+
+#
+if [ -n "$gitCommitSHA" ]
+then
+    # Set bundle build version
     echo "#define GIT_COMMIT_SHA $gitCommitSHA" >> $PROJECT_TEMP_DIR/infoplist.prefix
+    # Tell Xcode to preprocess Info.plist
+    touch $PROJECT_DIR/Info.plist
```

Comment:
@bbodenmiller see https://github.com/OneBusAway/onebusaway-iphone/pull/259#discussion_r25144812 — the error suggests some path isn't properly escaping the space in your disk's name.